l3kernel/l3str-convert.dtx

```diff
@@ -186,6 +186,19 @@
 %   encoding. Instead, the \meta{false code} is performed.
 % \end{function}
 %
+% \section{Conversion by expansion (for PDF contexts)}
+%
+% A small number of expandable functions are provided for use in PDF string/name
+% contexts. These \emph{assume UTF-8} and \emph{no escaping} in the input.
+%
+% \begin{function}[EXP]{\str_convert_pdfname:n}
+%   \begin{syntax}
+%     \cs{str_convert_pdfname:n} \meta{string}
+%   \end{syntax}
+%   As \cs{str_set_convert:Nnnn}, converts the \meta{string} on a byte-by-byte
+%   basis with non-ASCII codepoints  escaped using hashes.
+% \end{function}
+%
```

Comment:
I'm wondering whether there could be a need for more generally `\str_convert:nnn`, which is probably doable now that we have `e`-expansion.

Reply:
@blefloch  we have been discussing this, on the whole I don't think that this is a pressing issue. names pop up in various arguments, so being able to expand them is quite usefull, but other strings can normally be saved and handled later.  More needed is an utf8/xmp conversion/escaping for the xmp-metadata.